jib-core/src/test/java/com/google/cloud/tools/jib/registry/RegistryClientTest.java

```diff
@@ -211,6 +212,37 @@ public void testConfigureBasicAuth()
         registry.getInputRead(), CoreMatchers.containsString("Authorization: Basic dXNlcjpwYXNz"));
   }
 
+  @Test
+  public void testPullManifest()
+      throws IOException, InterruptedException, GeneralSecurityException, URISyntaxException,
+          RegistryException {
+    String manifestResponse =
+        "HTTP/1.1 200 OK\nContent-Length: 307\n\n{\n"
+            + "    \"schemaVersion\": 2,\n"
+            + "    \"mediaType\": \"application/vnd.docker.distribution.manifest.v2+json\",\n"
+            + "    \"config\": {\n"
+            + "        \"mediaType\": \"application/vnd.docker.container.image.v1+json\",\n"
+            + "        \"size\": 7023,\n"
+            + "        \"digest\": \"sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7\"\n"
+            + "    }\n"
+            + "}";
+
+    registry = new TestWebServer(false, Arrays.asList(manifestResponse), 1);
+    RegistryClient registryClient = createRegistryClient(null);
+    ManifestAndDigest<?> manifestAndDigest = registryClient.pullManifest("image-tag");
+
+    // verify details of the returned manifest
+    V22ManifestTemplate manifest = (V22ManifestTemplate) manifestAndDigest.getManifest();
+    Assert.assertEquals(2, manifest.getSchemaVersion());
+    Assert.assertEquals(
+        "application/vnd.docker.distribution.manifest.v2+json", manifest.getManifestMediaType());
+    Assert.assertEquals(
+        "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+        manifest.getContainerConfiguration().getDigest().toString());
+    Assert.assertEquals(7023, manifest.getContainerConfiguration().getSize());
+    Assert.assertTrue(manifestAndDigest.getManifest() instanceof V22ManifestTemplate);
+  }
```

Comment:
As I said before, could you also verify `manifestAndDigest.getDigest().toString()`?

Reply:
The manifest file itself doesn't contain a digest in this instance.The digest is in the config file so thats why I used `manifest.getContainerConfiguration().getDigest().toString())`.

Reply:
Oh! There's another test I forget to mention. Testing if `pullManifest()` is hitting the correct registry API endpoint for fetching a manifest.

`TestWebServer` has `getInputRead()`, which will return all the input `pullManifest()` has sent through the low-level HTTP protocol. If you check the return value of `TestWebServer.getInputRead()`, it will contain a line like this:
`GET /v2/foo/bar/manifests/image-tag ...`. So you can have

```java
Assert.assertThat(registry.getInputRead(), CoreMatchers.containsString("GET ..."));
```

Reply:
`manifest.getContainerConfiguration().getDigest().toString()` is the SHA256 hash of the test JSON input. (For example, if you change the content of the JSON, say, `"size": 7023` to `"size": 98`, the hash will change as well.) It's worth checking if `pullManifest()` is correctly computing the digest.

Reply:
Done